### DISCUSSION TITLE
Testing trading logic

### DISCUSSION THREAD

**Discussion by TradingData684** [erdewit/ib_insync#590] [General]
Hi fellow traders,

Is there a possibility to get some random data of the TWS during off hours or during the weekend? I dont want to wait until the market opens to develop my strategies and test them. So is there a way to get some random or simulated data when the markets are not opened?

Kind regards and thanks in advance :)

**Comment by jlixfeld** [comment]
There's historical data that is available 24/7.

**Reply by TradingData684** [reply]
Thats good to know. Where can I find/enable that?

**Comment by jlixfeld** [comment]
The [docs](https://ib-insync.readthedocs.io/)

**Reply by TradingData684** [reply]
well thanks for the answer...

**Comment by erdewit** [comment]
There used to be the edemo account with fake realtime data but it has unfortunately been discontinued by IB.

**Reply by TradingData684** [reply]
Thanks, that is the feature I was looking for. Sad to here that it is not available anymore. Is there any other alternative, since IB decided to cancel it?

**Comment by carlostipton1319** [comment]
Yes, it is possible to access simulated or historical market data during off hours or weekends to develop and test trading strategies. Many trading platforms and data providers offer simulated trading [crypto revolution](https://cryptorealtrading.com/the-crypto-revolution-and-the-need-for-real-time-data/) environments or historical data that you can use to backtest your strategies.
